Source/CarthageKitTests/Fakes/FakeURLSession.swift

```diff
@@ -0,0 +1,41 @@
+import Foundation
+
+final class FakeDownloadTask: NSURLSessionDownloadTask {
+	var started = false
+	override func resume() {
+		self.started = true
+	}
+
+	var cancelled = false
+	override func cancel() {
+		self.cancelled = true
+	}
+}
+
+final class FakeDataTask: NSURLSessionDataTask {
+	var started = false
+	override func resume() {
+		self.started = true
+	}
+
+	var cancelled = false
+	override func cancel() {
+		self.cancelled = true
+	}
+}
+
+final class FakeURLSession: NSURLSession {
```

Comment:
The docs for NSURLSession don't mention anything about subclassing (though the NSURLSession.h header does mention that the `init(configuration:)` and `init(configuration:delegate:delegateQueue:)` methods aren't inheritable), and I've been using this method for faking out the network since NSURLSession was introduced.

Additionally, this allowed me to re-use the existing `carthage_downloadWithRequest()` and `rac_dataWithRequest()` methods without making any modifications to them.